bee-node/src/lib.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+#![recursion_limit = "256"]
```

Comment:
that needed to be included because the `loop` in `run_loop` got more branches. You also see it in `bee-network` and `bee-protocol` (I think)

Reply:
It's not in protocol. Can you try without ? I'm not entirely sure it's needed.

Reply:
It's needed when your macros (probably select) gets heavy but I remember making it short in node 🤔

Reply:
I tried also in network by having each select variant just calling a function. I don't know a way of getting rid of it. It annoys me too.

Reply:
I usually delegate the whole thing to a function so that select just only ever calls it

Reply:
But I usually only have a listener and a shutdown

Reply:
yea, in the network I had to select over commands and events and the shutdown. I put as much as I could into functions which is a good thing anyway, but I couldn't get it to compile without this `recursion_limit` attribute.

Reply:
btw. it's not just us having issues with this attribute:
https://github.com/yewstack/yew/issues/513

The only solution seems to be to just increase the limit if need be.

Reply:
But are we actually having the issue on the node though ? I remember fixing it and being able to remove the attribute.

Reply:
yea, if I remove the attribute it will not compile. you probably were able to fix it because you introduced the functions, but with the new shutdown branch it again needs this extended limit.

Reply:
I basically only added this:
```rust
shutdown = shutdown => {
    break;
}
```
which is really not much! :smile:

Reply:
Still, it looks like it could be extracted in a function to avoid adding the attribute 🤔

Reply:
there is nothing to extract anymore :smiley:

Reply:
I could only outsource the `break` statement into a function, but what's the point. haha

Reply:
I was speaking about the match, it can easily be extracted and it would make sense.

Reply:
yup, you're right. It's extracted into 
```rust
#[inline]
    async fn handle_event(&mut self, event: Option<Event>) { ... }
```
that get's rid of the `recursion_limit` attribute

Reply:
do you think the `Option` should be handled outside? My feeling says 'yes'

Reply:
yea, only the `match` now :smile:

Reply:
My feeling says yes as well :)